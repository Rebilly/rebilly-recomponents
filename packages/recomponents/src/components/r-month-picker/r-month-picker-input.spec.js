import {mount, shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RMonthPickerInput from './r-month-picker-input.vue';
import RMonthPicker from './r-month-picker.vue';
import Vue from 'vue';

describe('r-month-picker-input.vue', () => {
    it('should render Wrapper and match snapshot', () => {
        const wrapper = shallowMount(RMonthPickerInput, {});
        expect(wrapper).toMatchSnapshot();
    });
    it('should render via SSR and match snapshot', async () => {
        const wrapper = await renderToString(RMonthPickerInput, {});
        expect(wrapper).toMatchSnapshot();
    });
    it('should select month and year if default values are set', async () => {
        const wrapper = await shallowMount(RMonthPickerInput, {
            propsData: {
                value: {monthIndex: 11, year: 2021},
            },
        });

        expect(wrapper.find('.r-month-picker-input').exists()).toBeTruthy();
        expect(wrapper.find('.r-month-picker-input').text()).toEqual('December, 2021');
    });
    it('should show month picker on click', async () => {
        const wrapper = await mount(RMonthPickerInput, {
            components: {RMonthPicker},
            propsData: {
                value: {monthIndex: 11, year: 2021},
            },
        });


        wrapper.find('.r-month-picker-input').trigger('click');

        await Vue.nextTick();

        expect(wrapper.find('.r-month-picker-month-selected').exists()).toBeTruthy();
        expect(wrapper.find('.r-month-picker-month-selected').text()).toEqual('December');

        const months = await wrapper.findAll('.r-month-picker-month');
        expect(months.at(0).text()).toEqual('January');
        months.at(0).trigger('click');

        await Vue.nextTick();

        expect(wrapper.emitted().input).toBeTruthy();
        expect(wrapper.emitted().input).toEqual([[{monthIndex: 0, year: 2021}]]);
        expect(wrapper.find('.r-month-picker-input').text()).toEqual('January, 2021');
    });
    it('should show placeholder if no month is selected', async () => {
        const placeholder = 'No month selected';
        const wrapper = await mount(RMonthPickerInput, {
            components: {RMonthPicker},
            propsData: {
                value: {monthIndex: 11, year: 2021},
                placeholder,
                clearable: true,
            },
        });

        wrapper.find('.r-month-picker-input').trigger('click');

        await Vue.nextTick();

        expect(wrapper.find('.r-month-picker-month-selected').exists()).toBeTruthy();
        expect(wrapper.find('.r-month-picker-month-selected').text()).toEqual('December');

        wrapper.find('.r-month-picker-month-selected').trigger('click');
        expect(wrapper.emitted().clear).toBeTruthy();

        await Vue.nextTick();

        expect(wrapper.find('.r-month-picker-input').text()).toEqual(placeholder);
    });
});
