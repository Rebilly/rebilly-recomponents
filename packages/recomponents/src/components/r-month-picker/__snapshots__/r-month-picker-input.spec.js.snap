// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-month-picker-input.vue should render Wrapper and match snapshot 1`] = `
<r-popper-stub esctohide="true" autohide="true" globalautohide="true" offset="4" margin="0,0,0,0" direction="horizontal" position="bottomEnd" slidefrom="fade" duration="0.2">
  <r-icon-button-stub type="default" size="regular" capitalizefirstletter="true" class="r-month-picker-input"><span>June, 2020</span>
    <r-icon-stub icon="calendar" class="r-date-range-calendar-icon"></r-icon-stub>
  </r-icon-button-stub>
  <div class="r-popover r-month-picker-input-container">
    <div class="r-popover-control">
      <div width="300px" class="r-popover-content r-popover-menu">
        <r-month-picker-stub value="[object Object]"></r-month-picker-stub>
      </div>
    </div>
  </div>
</r-popper-stub>
`;

exports[`r-month-picker-input.vue should render via SSR and match snapshot 1`] = `
<div class="r-popper"><button class="r-button r-month-picker-input r-button-type-default r-button-size-regular r-has-icon-left"><svg class="r-icon r-icon-20 r-date-range-calendar-icon">
      <use xlink:href="#icon-calendar" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
    </svg>
    <div role="button" class="r-capitalize-first-letter button"><span>June, 2020</span> </div>
  </button>
  <div class="r-popper-content-wrapper">
    <transition-stub name="fade">
      <!---->
    </transition-stub>
  </div>
</div>
`;
